stdlib/public/core/String.swift

```diff
@@ -321,11 +321,10 @@ extension String {
     let (stringBufferOptional, _) =
         _StringBuffer.fromCodeUnits(input, encoding: encoding,
             repairIllFormedSequences: false)
-    if let stringBuffer = stringBufferOptional {
-      return String(_storage: stringBuffer)
-    } else {
+    guard let stringBuffer = stringBufferOptional else {
       return nil
     }
+    return String(_storage: stringBuffer)
```

Comment:
I think this would actually be better as `return stringBufferOptional.map { String(_storage: $0) }`.